// @kinde-oss/kinde-auth-react/dist/utils.mjs
import { j as s, q as t, I as r, i as o, d as g, V as i, B as n, N as c, r as l, H as S, a as m, L as h, s as u, v as P, T as f, R as p, o as A, t as R, e as T, n as d, K as U, x as C, C as k, p as y, l as I, M as v, u as x, z, f as w, D, g as E, m as F, E as L, w as M, k as O, h as b, F as K, G as j, S as q, J as B, O as G, $ as H, P as J, _ as N, y as Q, Q as V, b as _, c as $ } from "./index-DjNR3H6H.js";
export {
  s as ChromeStore,
  t as ExpoSecureStore,
  r as IssuerRouteTypes,
  o as LocalStorage,
  g as MemoryStorage,
  i as PortalPage,
  n as ProfilePage,
  c as PromptTypes,
  l as RefreshType,
  S as Scopes,
  m as StorageKeys,
  h as base64UrlEncode,
  u as checkAuth,
  P as clearActiveStorage,
  f as clearInsecureStorage,
  p as clearRefreshTimer,
  A as exchangeAuthCode,
  R as extractAuthResults,
  T as frameworkSettings,
  d as generateAuthUrl,
  U as generatePortalUrl,
  C as generateProfileUrl,
  k as generateRandomString,
  y as getActiveStorage,
  I as getClaim,
  v as getClaims,
  x as getCurrentOrganization,
  z as getDecodedToken,
  w as getFlag,
  D as getInsecureStorage,
  E as getPermission,
  F as getPermissions,
  L as getRawToken,
  M as getRoles,
  O as getUserOrganizations,
  b as getUserProfile,
  K as hasActiveStorage,
  j as hasInsecureStorage,
  q as isAuthenticated,
  B as isCustomDomain,
  G as mapLoginMethodParamsForUrl,
  H as refreshToken,
  J as sanitizeUrl,
  N as setActiveStorage,
  Q as setInsecureStorage,
  V as setRefreshTimer,
  _ as splitString,
  $ as storageSettings
};
