// @kinde/js-utils/dist/js-utils.js
import { O as s, E as t, V as r, P as o, N as g, T as i, U as n, R as c, W as l, Q as S, a as m, c as h, i as u, I as P, M as f, l as p, h as A, e as R, n as T, f as U, p as d, o as C, g as k, G as y, q as I, r as v, t as x, v as z, w, K as D, y as E, z as F, u as L, B as M, A as O, x as b, H as K, L as j, C as q, j as B, m as G, D as H, d as J, F as N, J as Q, k as V, s as W, b as X } from "./main-CMi82f2p.js";
export {
  s as ChromeStore,
  t as ExpoSecureStore,
  r as IssuerRouteTypes,
  o as LocalStorage,
  g as MemoryStorage,
  i as PortalPage,
  n as ProfilePage,
  c as PromptTypes,
  l as RefreshType,
  S as Scopes,
  m as StorageKeys,
  h as base64UrlEncode,
  u as checkAuth,
  P as clearActiveStorage,
  f as clearInsecureStorage,
  p as clearRefreshTimer,
  A as exchangeAuthCode,
  R as extractAuthResults,
  T as frameworkSettings,
  U as generateAuthUrl,
  d as generatePortalUrl,
  C as generateProfileUrl,
  k as generateRandomString,
  y as getActiveStorage,
  I as getClaim,
  v as getClaims,
  x as getCurrentOrganization,
  z as getDecodedToken,
  w as getFlag,
  D as getInsecureStorage,
  E as getPermission,
  F as getPermissions,
  L as getRawToken,
  M as getRoles,
  O as getUserOrganizations,
  b as getUserProfile,
  K as hasActiveStorage,
  j as hasInsecureStorage,
  q as isAuthenticated,
  B as isCustomDomain,
  G as mapLoginMethodParamsForUrl,
  H as refreshToken,
  J as sanitizeUrl,
  N as setActiveStorage,
  Q as setInsecureStorage,
  V as setRefreshTimer,
  W as splitString,
  X as storageSettings
};
